src/request/sign-transaction/SignTransaction.js

```diff
@@ -11,29 +72,44 @@ class SignTransaction { // eslint-disable-line no-unused-vars
      * @returns {Promise<SignedTransactionResult>}
      * @protected
      */
-    async _signTx(txRequest, passphraseOrPin) {
-        if (txRequest.type !== TransactionType.BASIC) {
-            throw new Error('Not yet implemented');
+    async _doSignTransaction(txRequest, passphraseOrPin) {
+        if (txRequest.type === TransactionType.BASIC) {
+            // eslint-disable-next-line object-curly-newline
+            const { value, fee, recipient, signer, validityStartHeight } = txRequest;
+
+            const key = await KeyStore.instance.get(signer, passphraseOrPin);
+            const tx = key.createTransaction(recipient, value, fee, validityStartHeight);
+
+            const signatureProof = Nimiq.SignatureProof.unserialize(new Nimiq.SerialBuffer(tx.proof));
+
+            return {
+                type: TransactionType.BASIC,
+                sender: tx.sender.toUserFriendlyAddress(),
+                signerPubKey: signatureProof.publicKey.serialize(),
+                recipient: tx.recipient.toUserFriendlyAddress(),
+                value: Nimiq.Policy.satoshisToCoins(tx.value),
+                fee: Nimiq.Policy.satoshisToCoins(tx.fee),
```

Comment:
why not leave them as satoshis?

Reply:
It's a basic API decision. Originally we designed the API to be as user-friendly as possible, thus it always used NIM for values. We can change this, but then need to make it the same for all interfaces. I for one am in favour of using NIM for values, but it can be a majority decision.

Reply:
One argument for using satoshis as unit is that if one accidentally sends 1000 satoshis when his intention was to send 1000 NIM, it's not the same problem as the other way around.

Reply:
However, I think coins can be a vary valid choice too. We should decide this apart from this PR.